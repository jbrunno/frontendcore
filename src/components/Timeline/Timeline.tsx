import React, { Ref } from 'react';
import {
  Timeline as MuiTimeline,
  TimelineProps as MuiTimelineProps,
} from '@mui/lab';

export type TimelineProps = MuiTimelineProps & {
  ref?: Ref<HTMLUListElement>;
};

export function Timeline(props: TimelineProps) {
  return <MuiTimeline {...props} />;
}

export default Timeline;
