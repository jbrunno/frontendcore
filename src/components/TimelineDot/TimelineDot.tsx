import React from 'react';
import {
  TimelineDot as MuiTimelineDot,
  TimelineDotProps as MuiTimelineDotProps,
} from '@mui/lab';

export type TimelineDotProps = MuiTimelineDotProps;

export function TimelineDot(props: TimelineDotProps) {
  return <MuiTimelineDot {...props} />;
}

export default TimelineDot;
