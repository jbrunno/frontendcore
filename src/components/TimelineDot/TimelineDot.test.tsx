import React from 'react';
import { render } from '@test-utils/component-testing';
import { TimelineDot } from './TimelineDot';
import Timeline from '../Timeline/Timeline';

test('renders the TimelineDot', () => {
  const { container } = render(
    <Timeline>
      <TimelineDot />
    </Timeline>,
  );
  expect(container).toMatchSnapshot();
});
