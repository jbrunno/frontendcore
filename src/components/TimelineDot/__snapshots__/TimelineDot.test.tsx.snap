// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TimelineDot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  border-style: solid;
  border-width: 2px;
  padding: 4px;
  border-radius: 50%;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  margin: 11.5px 0;
  border-color: transparent;
  color: #fafafa;
  background-color: #bdbdbd;
}

<div>
  <ul
    class="MuiTimeline-root MuiTimeline-positionRight emotion-0"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey emotion-1"
    />
  </ul>
</div>
`;
